components/security/authorization/src/test/java/org/phenotips/security/authorization/internal/DefaultAuthorizationServiceTest.java

```diff
@@ -76,132 +88,104 @@ public void setupMocks() throws Exception
         // FIXME This should be done in MockitoComponentMockingRule automatically
         MockitoAnnotations.initMocks(this);
         resetMocks();
+        ReflectionUtils.setFieldValue(this.mocker.getComponentUnderTest(), "modules", this.modules);
     }
 
     @Test
     public void defaultDecisionIsDeny() throws ComponentLookupException
     {
-        for (ComponentDescriptor<?> cd : this.mocker.getComponentDescriptorList((Type) AuthorizationModule.class)) {
-            this.mocker.unregisterComponent(cd);
-        }
+        doReturn(new LinkedList<>()).when(this.modules).get();
         Assert.assertFalse(this.mocker.getComponentUnderTest().hasAccess(this.user, this.access, this.document));
     }
 
     @Test
     public void moduleDecisionIsUsed() throws Exception
     {
-        this.mocker.registerComponent(AuthorizationModule.class, "low", this.lowPriorityModule);
+        this.moduleList = Collections.singletonList(this.moduleOne);
+        doReturn(this.moduleList).when(this.modules).get();
 
-        when(this.lowPriorityModule.hasAccess(this.user, this.access, this.document)).thenReturn(true);
+        when(this.moduleOne.hasAccess(this.user, this.access, this.document)).thenReturn(true);
         Assert.assertTrue(this.mocker.getComponentUnderTest().hasAccess(this.user, this.access, this.document));
 
-        when(this.lowPriorityModule.hasAccess(this.user, this.access, this.document)).thenReturn(false);
+        when(this.moduleOne.hasAccess(this.user, this.access, this.document)).thenReturn(false);
         Assert.assertFalse(this.mocker.getComponentUnderTest().hasAccess(this.user, this.access, this.document));
 
-        when(this.lowPriorityModule.hasAccess(this.user, this.access, this.document)).thenReturn(null);
+        when(this.moduleOne.hasAccess(this.user, this.access, this.document)).thenReturn(null);
         Assert.assertFalse(this.mocker.getComponentUnderTest().hasAccess(this.user, this.access, this.document));
     }
 
     @Test
     public void modulesAreCascadedUntilNonNullIsReturned() throws Exception
     {
-        this.mocker.registerComponent(AuthorizationModule.class, "low", this.lowPriorityModule);
-        this.mocker.registerComponent(AuthorizationModule.class, "medium", this.mediumPriorityModule);
-        this.mocker.registerComponent(AuthorizationModule.class, "high", this.highPriorityModule);
+        this.moduleList = Arrays.asList(this.moduleOne, this.moduleTwo, this.moduleThree);
+        doReturn(this.moduleList).when(this.modules).get();
```

Comment:
Using `ReflectionUtils.setFieldValue` should be avoided, but I don't yet see a different way of testing this.